src/prefect/events/clients.py

```diff
@@ -195,3 +210,160 @@ async def _emit(self, event: Event) -> None:
                     # a standard load balancer timeout, but after that, just take a
                     # beat to let things come back around.
                     await asyncio.sleep(1)
+
+
+class PrefectCloudEventSubscriber:
+    """
+    Subscribes to a Prefect Cloud event stream, yielding events as they occur.
+
+    Example:
+
+        from prefect.events.clients import PrefectCloudEventSubscriber
+        from prefect.events.filters import EventFilter, EventNameFilter
+
+        filter = EventFilter(event=EventNameFilter(prefix=["prefect.flow-run."]))
+
+        async with PrefectCloudEventSubscriber(api_url, api_key, filter) as subscriber:
+            async for event in subscriber:
+                print(event.occurred, event.resource.id, event.event)
+
+    """
+
+    _websocket: Optional[WebSocketClientProtocol]
+    _filter: "EventFilter"
+    _seen_events: TTLCache[UUID, bool]
+
+    def __init__(
+        self,
+        api_url: str,
+        api_key: str,
```

Comment:
Can we default these to read from the current settings if they aren't provided?

Reply:
Ah that's a good idea, I'll do that!